File: DevExpress-Demo/VB/MVVMDemo.Wpf/Modules/DXCommand/MultipleCallsViewModel.vb
Imports DevExpress.Mvvm
Imports System.Windows

Namespace MVVMDemo.DXCommandDemo

    Public Class MultipleCallsViewModel
        Inherits BindableBase

        Public Property IsReadonly As Boolean
            Get
                Return GetValue(Of Boolean)()
            End Get

            Set(ByVal value As Boolean)
                SetValue(value)
            End Set
        End Property

        Public Property FileName As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public Sub Open()
            MessageBox.Show(String.Format("Action: Open {0}", FileName))
        End Sub

        Public Function CanOpen() As Boolean
            Return Not String.IsNullOrEmpty(FileName)
        End Function

        Public Sub Clear()
            FileName = Nothing
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/MVVMDemo.Wpf/Modules/DXCommand/MultipleCallsViewModel.cs
using DevExpress.Mvvm;
using System.Windows;

namespace MVVMDemo.DXCommandDemo {
    public class MultipleCallsViewModel : BindableBase {
        public bool IsReadonly {
            get { return GetValue<bool>(); }
            set { SetValue(value); }
        }

        public string FileName {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }

        public void Open() {
            MessageBox.Show(string.Format("Action: Open {0}", FileName));
        }
        public bool CanOpen() {
            return !string.IsNullOrEmpty(FileName);
        }

        public void Clear() {
            FileName = null;
        }
    }
}
